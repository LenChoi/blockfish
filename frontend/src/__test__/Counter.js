import React, { useState, useCallback } from 'react';

const HookCounter = () => {
  const [number, setNumber] = useState(0);
  const onIncrease = useCallback(() => {
    setNumber(number + 1);
  }, [number]);
  const onDecrease = useCallback(() => {
    setNumber(number - 1);
  }, [number]);

  return (
    <div>
      <h2>{number}</h2>
      <button type="button" onClick={onIncrease}>
        +1
      </button>
      <button type="button" onClick={onDecrease}>
        -1
      </button>
    </div>
  );
};

export default HookCounter;
