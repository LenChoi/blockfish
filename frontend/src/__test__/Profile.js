import React from 'react';

const Profile = ({ username, name }) => (
  <div>
    <b>{username}</b>&nbsp;
    <span>({name})</span>
  </div>
);

export default Profile;
