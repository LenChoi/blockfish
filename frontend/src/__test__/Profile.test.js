import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';
import Counter from './Counter';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Profile username="velopert" name="김민준" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders username and name', () => {
    const wrapper = mount(<Profile username="velopert" name="김민준" />);

    expect(wrapper.props().username).toBe('velopert');
    expect(wrapper.props().name).toBe('김민준');
  });

  it('increases', () => {
    const wrapper = mount(<Counter />);
    const plusButton = wrapper.findWhere(
      (node) => node.type() === 'button' && node.text() === '+1',
    );
    plusButton.simulate('click');
    plusButton.simulate('click');

    const number = wrapper.find('h2');

    expect(number.text()).toBe('2');
  });
  it('decreases', () => {
    const wrapper = mount(<Counter />);
    const decreaseButton = wrapper.findWhere(
      (node) => node.type() === 'button' && node.text() === '-1',
    );
    decreaseButton.simulate('click');
    decreaseButton.simulate('click');

    const number = wrapper.find('h2');

    expect(number.text()).toBe('-2');
  });
});
