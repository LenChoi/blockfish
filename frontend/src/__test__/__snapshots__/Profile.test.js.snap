// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="김민준"
  username="velopert"
>
  <div>
    <b>
      velopert
    </b>
     
    <span>
      (
      김민준
      )
    </span>
  </div>
</Profile>
`;
